Update Haskell package set to LTS 16.2 (plus other fixes)

peti opened the pull request:
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on Friday, 2020-06-26 20:00 +02:00. You can watch this live on Twitch at https://www.twitch.tv/peti343. In addition to the chat features offered by Twitch, there is also a voice conference at https://discord.gg/YTEa3XR that viewers can use to chat with me and with each other.

sternenseemann committed:
haskellPackages.utc: unbreak

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-5-g7c30584 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/a4534198c9406045f79eac25df59dedbe4524826.

cdepillabout committed:
Merge pull request #91103 from sternenseemann/unbreak-utc

haskellPackages.utc: unbreak

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-5-g7c30584 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/fa1cee3827410c77df71b292a5a07d61aa5282ef.

maralorn committed:
haskellPackages.reflex-dom and deps: Fix patch urls

The previously used PR specific urls do cease to exist after a force
push to the PR branch.

Also manually enable hydra builds for reflex-dom-core until upstream
provides a better fix.

balsoft committed:
haskellPackages.gi-gtk-declarative: fix build by disabling tests

gi-gtk-declarative requires an X11 display in its test suite.
X11 display is not available inside nix build sandbox, which makes tests fail.

Solution: disable tests for gi-gtk-declarative.

maralorn committed:
hackagePackages.hnix: Fix build

hnix 0.9.0 does not provide an executable for ghc <8.10 anymore.
So we need to disable optparse-applicative completion generation for all other versions.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-5-g7c30584 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/db7bf3bb1b774657b089749fe414002befb41d99.

cdepillabout committed:
Merge pull request #91233 from maralorn/fix-reflex-patches

[haskell-updates] haskellPackages.reflex-dom and deps: Fix patch urls

cdepillabout committed:
Merge pull request #91237 from balsoft/fix-gi-gtk-declarative

haskellPackages.gi-gtk-declarative: fix build by disabling tests

Anton-Latukha committed:
statistics: clean-up

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
binary-instances: update override description

Previously mentioned issue closed as duplicate of an older one.

Issue waits upon QuickCheck release that includes a patch.

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
nixfmt: update override description

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
rib: update override description

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
xmobar: update patch description

M  pkgs/development/haskell-modules/configuration-common.nix

maralorn committed:
haskellPackages.jsaddle-warp: Fix build

maralorn committed:
haskellPackages: Fix gtk2hs packages

glib and friends have been bumped to 0.13.8.1 so we can drop a patch
picked from that version.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-5-g7c30584 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/c69592683fdd0265a0918fdda036319142f8f735.

cdepillabout committed:
Merge pull request #91315 from maralorn/fix-jsaddle-warp

[haskell-updates] haskellPackages.jsaddle-warp: Fix build

cdepillabout committed:
Merge pull request #91251 from maralorn/fix-hnix

[haskell-updates] hackagePackages.hnix: Disable completion to fix build

Anton-Latukha committed:
pantry: free from override and upstreamed patch

M  pkgs/development/haskell-modules/configuration-common.nix

Anton-Latukha committed:
feed: free from override

M  pkgs/development/haskell-modules/configuration-common.nix

domenkozar committed:
haskell.compiler.ghc865-binary: strip to fix aarch64

maralorn committed:
haskellPackages.stan: Fix build

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-5-g7c30584 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/04f759deb65577ab8494cc2f95c7ba2def2ca826.

svmhdvn committed:
haskellPackages.hsimport: unbreak

cdepillabout committed:
Merge pull request #91319 from maralorn/fix-glib

[haskell-updates] haskellPackages: Fix gtk2hs packages

cdepillabout committed:
Merge pull request #91388 from svmhdvn/patch-1

haskellPackages.hsimport: unbreak

cdepillabout committed:
Merge pull request #91312 from Anton-Latukha/upd-Haskell-configuration-common-01

Haskell: configuration-common: updates to overrides

cdepillabout committed:
Merge pull request #91383 from maralorn/fix-stan

[haskell-updates] haskellPackages.stan: Fix build

svmhdvn committed:
haskellPackages.rdf4h: dontCheck

svmhdvn committed:
Update configuration-common.nix

svmhdvn committed:
haskellPackages: remove obsolete overrides

esqueleto: -dontHaddock
blank-canvas: -dontCheck
tdigest: -doJailbreak

svmhdvn committed:
haskellPackages: remove obsolete overrides

classy-prelude-yesod: -dontHaddock
rank2classes: -dontCheck
HaTeX: -dontCheck

svmhdvn committed:
haskellPackages: remove obsolete overrides

euler: -dontCheck
snappy-framing: -dontHaddock
vector: -dontCheck

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-5-g7c30584 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/cae7d034d7ddf134f3d0eb99470b272350ac535b.

svmhdvn committed:
haskellPackages: remove obsolete overrides

stm-conduit: -dontCheck
asn1-encoding: -dontCheck
geodetics: -dontCheck

cdepillabout committed:
Merge pull request #91420 from svmhdvn/patch-3

haskellPackages.rdf4h: dontCheck

peti committed:
LTS Haskell 16.2

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-5-g7c30584 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/711b59b900ee0b267ebb86e41dbc83ad7fb69629.

peti committed:
all-cabal-hashes: update to Hackage at 2020-06-25T08:18:20Z

peti committed:
Merge remote-tracking branch 'origin/master' into haskell-updates.

peti committed:
haskell-repline: update override for ghc-8.10.x to the latest version

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-5-g7c30584 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/711b59b900ee0b267ebb86e41dbc83ad7fb69629.

cdepillabout committed:
Merge pull request #91502 from svmhdvn/overrides-1

haskellPackages: cleaning up obsolete overrides (1/x)

cdepillabout committed:
Merge pull request #91508 from svmhdvn/overrides-2

haskellPackages: cleaning up obsolete overrides (2/x)

cdepillabout committed:
Merge pull request #91513 from svmhdvn/overrides-3

haskellPackages: cleaning up obsolete overrides (3/x)

cdepillabout committed:
Merge pull request #91519 from svmhdvn/overrides-4

haskellPackages: cleaning up obsolete overrides (4/x)

sorki commented:
When testing updates with `hackage2nix` it seems to add `ConcurrentUtils` which depends on `assert` package and we get:

```nix
"ConcurrentUtils" = callPackage
    ({ mkDerivation, array, assert, atl, base, containers, extra
     , list-extras, monad-loops, monads-tf, parallel, profunctors
     , strict, time, vector
     }: ...)
```

which causes `unexpected ASSERT` error, even noticed by highlighter. I'm not quite sure what's the right way to fix something like this.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-6-g17421b6 from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/f94245ebd389a3f37c8c5ade8dd9faf4f288721b.

peti committed:
Merge remote-tracking branch 'origin/master' into haskell-updates.

peti commented:
@sorki,

> When testing updates with `hackage2nix` it seems to add `ConcurrentUtils` which depends on `assert` package and we get: [...]

I added a work-around for that a while ago: https://github.com/NixOS/cabal2nix/commit/b7c82dbe6d2bb571c92771385bdee6451b087b0f. It's not pretty, but it gets the job done.

peti committed:
hackage2nix: disable broken packages to fix evaluation errors

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.15.5-6-g17421b6-dirty from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/16740bbf7ab121fb84138c94018c4e6efbd7e372.

peti committed:
haskell-policeman: jailbreak with all compilers to fix the overspecified constraint on relude

peti committed:
hackage2nix: update list of broken packages

@kiwi: please note that matterhorn & Unique are both broken

peti committed:
hackage2nix: keep repline 0.3.x around for dhall

peti committed:
haskell-dhall: fix overrides for the 0.32 version

peti commented:
Next week, we'll merge https://github.com/NixOS/nixpkgs/pull/91759.
